storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -1445,9 +1464,10 @@ bool Redo_Log_Data_Manager::stop_at(lsn_t lsn, lsn_t checkpoint_lsn) {
   archived_log_monitor.stop();
 
   /* to ensure redo logs are not disabled during the backup, reopen the log
-  files to read HEADER */
-  if (!reopen_log_files(lsn)) {
-    xb::error() << "Cannot find file with LSN " << lsn;
+  files to read HEADER. */
+  if (opt_lock_ddl == false && archived_log_state == ARCHIVED_LOG_NONE &&
+      !reopen_log_files()) {
```

Comment:
Why are we removing the lsn, don't we want to ensure that we still have the file with the lsn we want to stop?
Also are we sure that `log_files_find_and_analyze` return something else rather than Log_files_find_result::FOUND_VALID_FILES when redo is disabled?

Can we have a test case that fails the backup at this part?

Reply:
@altmannmarcelo  Here we want to ensure redo is not disabled. and checking `log_files_find_and_analyze` is sufficient in case redo is disabled. yes, I have tested that. and there is already a test case `alter_instance_redo.sh` that fails the backup at this part.

Reply:
Regarding the test case - can you point which test case from `alter_instance_redo.sh`  covers this if condition?

Reply:
It is inside `case #4 check redo log is disabled during incremental backup`
Output from test result.
`2023-02-06T11:21:52.050353+05:30 0 [ERROR] [MY-013598] [InnoDB] Redo logging is disabled, cannot take consistent backup
2023-02-06T11:21:52.050366+05:30 0 [ERROR] [MY-011825] [Xtrabackup] Failed to open redo log files.
2023-02-06T11:21:52.050373+05:30 0 [ERROR] [MY-011825] [Xtrabackup] Error stopping copy thread at LSN 20619739`

Reply:
@rahulmalik87 do we grep for this output (Redo logging is disabled..)?